src/parser.rs

```diff
@@ -123,10 +123,9 @@ fn collapse_chunks(chunks: &[EncodedChunk]) -> Result<Vec<EncodedChunk>, Restore
     let mut mismatching = Vec::<(EncodedChunk, String)>::new();
     // Group by chunk id
     for (_chunk_id, mut duplicate_chunks) in &chunks.iter().group_by(|chunk| chunk.id) {
-        let reference_chunk = match duplicate_chunks.next() {
-            Some(v) => v,
-            None => panic!("Couldn't get 1 item per groupby"),
-        };
+        let reference_chunk = duplicate_chunks
+            .next()
+            .expect("Couldn't get 1 item per groupby");
```

Comment:
Oh yeah I forgot that this was a thing...
Good practice to have this kind of code smell debunked, thanks!